src/Tizen.Applications.Cion/Tizen.Applications/PayloadAsyncResult.cs

```diff
@@ -0,0 +1,129 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Tizen.Applications
+{
+    /// <summary>
+    /// A class to represent result of payload.
+    /// </summary>
+    /// <since_tizen> 9 </since_tizen>
+    public class PayloadAsyncResult : IDisposable
+    {
+        private readonly string LogTag = "Tizen.Cion";
+        internal PayloadAsyncResultSafeHandle _handle;
+
+        internal PayloadAsyncResult(PayloadAsyncResultSafeHandle handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Gets the result of payload.
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public PayloadAsyncResultCode Result
+        {
+            get
+            {
+                int code;
+                Interop.Cion.ErrorCode ret = Interop.CionPayloadAsyncResult.CionPayloadAsyncResultGetResult(_handle, out code);
+                if (ret != Interop.Cion.ErrorCode.None)
+                {
+                    return PayloadAsyncResultCode.Error;
+                }
+                return (PayloadAsyncResultCode)code;
+            }
+        }
+
+        /// <summary>
+        /// Gets the peer info of payload.
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public PeerInfo PeerInfo
+        {
+            get
+            {
+                IntPtr peer;
+                Interop.Cion.ErrorCode ret = Interop.CionPayloadAsyncResult.CionPayloadAsyncResultGetPeerInfo(_handle, out peer);
+                if (ret != Interop.Cion.ErrorCode.None)
+                {
+                    Log.Error(LogTag, "Fail to get peerinfo from the AsyncResult");
+                    return null;
+                }
+                return new PeerInfo(new PeerInfoSafeHandle(peer, true));
+            }
+        }
+
+        /// <summary>
+        /// Gets the payload id.
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public string PayloadId
+        {
+            get
+            {
+                Interop.Cion.ErrorCode ret = Interop.CionPayloadAsyncResult.CionPayloadAsyncResultGetPayloadID(_handle, out string payloadId);
+                if (ret != Interop.Cion.ErrorCode.None)
+                {
+                    Log.Error(LogTag, "Fail to get payload Id : " + ret);
+                    return "";
+                }
+                return payloadId;
+            }
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false;
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
```

Comment:
param is not described 
`/// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>`

Reply:
I added missing doxygen comment.